external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaProducer.scala

```diff
@@ -36,7 +36,7 @@ private[kafka010] object CachedKafkaProducer extends Logging {
   private type Producer = KafkaProducer[Array[Byte], Array[Byte]]
 
   private val cacheExpireTimeout: Long =
-    System.getProperty("spark.kafka.guava.cache.timeout", "10").toLong
+    System.getProperty("spark.kafka.guava.cache.timeout.minutes", "10").toLong
```

Comment:
don't we need to get this from `SparkEnv` by the way? I don't know if the properties get populated properly.
Also, adding `minutes` to the conf makes it kinda long right? I think we can also replace `guava` with `producer`. 
I think it may also be better to use [this](https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java#L213) so that we get rid of `minutes` and users can actually provide arbitrary durations (hours if they want). I think that's what we generally use for `duration` type confs.